src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1097,38 +1026,23 @@ create_domain (JNIEnv *env, jobjectArray runtimeApks, jstring assembly, jobject
 		monoFunctions.assembly_name_free (aname);
 	} else {
 		// Inflate environment from user app assembly
-		load_assembly (domain, env, assembly);
+		//		load_assembly (domain, env, assembly);
```

Comment:
Are we dropping that?

Reply:
Yep, assemblies are loaded fully on-demand now

Reply:
@grendello if assemblies are loaded on-demand now, we should check and make sure a scenario still works in Xamarin.Forms:
1. A custom renderer (or anything using their DI) is defined in a separate assembly than the app
2. Xamarin.Forms hits `AppDomain.GetAssemblies`: https://github.com/xamarin/Xamarin.Forms/blob/505a855cad39322c8b235281d71c35b70c9df129/Xamarin.Forms.Platform.Android/Forms.cs#L412-L415

I'm wondering if something would break here?

I have a test project you could try: https://github.com/jonathanpeppers/glidex/tree/master/glidex.forms.sample

Reply:
@jonathanpeppers I'll test it, thanks. It would be unfortunate of XF relied on `AppDomain.GetAssemblies` to work properly...

Reply:
@jonathanpeppers just tested - it works! :)